### PR TITLE
add flag to reset the score on content with only null results for ebs

### PR DESCRIPTION
**Pull request by AdamDM**
flag can be set to change logic in combiner to set score to 0 if all scorers returned null
ebs sport content registered to use this flag

### CODE REVIEW

File: src/main/java/org/atlasapi/equiv/EquivModule.java [hunk 1]
```diff
@@ -282,6 +283,34 @@ public boolean apply(Broadcast input) {
...
+                                new LookupWritingEquivalenceHandler<Item>(lookupWriter, acceptablePublishers),
+                                equivSummaryStore
+                        ),
+                        new ResultWritingEquivalenceHandler<Item>(equivalenceResultStore()),
+                        new EquivalenceSummaryWritingHandler<Item>(equivSummaryStore),
+                        MessageQueueingResultHandler.create(
+                                equivAssertDestination(), acceptablePublishers, lookupEntryStore
+                        )
+                )));
+    }
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/equiv/EquivModule.java]: lots of lines greater >100 characters in this method

File: src/main/java/org/atlasapi/equiv/EquivModule.java [hunk 2]
```diff
@@ -282,6 +283,34 @@ public boolean apply(Broadcast input) {
                 )
             )));
     }
+
+    private ContentEquivalenceUpdater.Builder<Item> ebsItemUpdater(Set<Publisher> acceptablePublishers, Set<? extends EquivalenceScorer<Item>> scorers) {
+
+        NullScoreAwareAveragingCombiner<Item> combiner = new NullScoreAwareAveragingCombiner<>();
+        combiner.setIgnoreNullScoringContent(true);
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/EquivModule.java]: Do not make a setter for this. This flag should be passed in the constructor and be `final`.

File: src/main/java/org/atlasapi/equiv/EquivModule.java (hunk 1 above)
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/EquivModule.java]: That

File: src/main/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombiner.java [hunk 3]
```diff
@@ -22,11 +22,19 @@
...
     
+    private boolean ignoreNullScoringContent;
+
     public static final <T extends Described> NullScoreAwareAveragingCombiner<T> get() {
         return new NullScoreAwareAveragingCombiner<T>();
     }
     
-    public NullScoreAwareAveragingCombiner() {}
+    public NullScoreAwareAveragingCombiner() {
+        ignoreNullScoringContent = false;
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombiner.java]: `ignoreNullScoringCandidate`

File: src/main/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombiner.java [hunk 4]
```diff
@@ -78,6 +86,11 @@ private void addCount(final Map<T, Integer> counts, Entry<T, Score> equivScore)
             public Score transformEntry(T key, Score value) {
                 if (value.isRealScore()) {
                     Integer count = publisherCounts.get(key.getPublisher());
+
+                    if (ignoreNullScoringContent && count == 1) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombiner.java]: Add javadoc explaining that if `count == 1` then only the only score we have must be from the generator and no scorers have returned a non-null score.

File: src/main/java/org/atlasapi/equiv/update/ContentEquivalenceUpdater.java [hunk 5]
```diff
@@ -25,30 +25,83 @@
...
 public class ContentEquivalenceUpdater<T extends Content> implements EquivalenceUpdater<T> {
 
-    public static final <T extends Content> Builder<T> builder() {
+    public static <T extends Content> Builder<T> builder() {
         return new Builder<T>();
     }
     
-    public static final class Builder<T extends Content> {
+    private final ScoredEquivalentsMerger merger = new ScoredEquivalentsMerger();
+    private final Function<ScoredCandidates<T>, Iterable<T>> extractCandidates = new Function<ScoredCandidates<T>, Iterable<T>>() {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/update/ContentEquivalenceUpdater.java]: Line is too long

File: src/main/java/org/atlasapi/equiv/update/ContentEquivalenceUpdater.java [hunk 6]
```diff
@@ -25,30 +25,83 @@
 
 public class ContentEquivalenceUpdater<T extends Content> implements EquivalenceUpdater<T> {
 
-    public static final <T extends Content> Builder<T> builder() {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/equiv/update/ContentEquivalenceUpdater.java]: I presume all the changes in this file are just methods being moved around and there are no functional changes.

Refactoring-only changes like this should be done in a separate commit from functional changes.

File: src/test/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombinerTest.java [hunk 7]
```diff
@@ -115,4 +115,36 @@ public void testCombiningScoresForSamePublisherAveragesOverHighestNumberOfSource
         assertEquals(Score.valueOf(2.0), combined.candidates().get(equivalent5));
         
     }
+
+    @Test
+    public void testCombineWithIgnoreNullFlagOn() {
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombinerTest.java]: Do not prefix tests with `test`. Format should be `ifWeDoXThenWeExpectY`

Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/results/combining/NullScoreAwareAveragingCombinerTest.java]: This should be multiple tests:

`combineSingleNonNullScoreWithIgnoreNullScoringCandidatesFlagOffReturnsScore`
`combineSingleNonNullScoreWithIgnoreNullScoringCandidatesFlagOnReturnsNull`

etc.
